http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -201,17 +219,16 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
       ) // TODO: update this to support Azure
       samVisibleDisksOpt <- NonEmptyList.fromList(diskAndProjects).traverse { ds =>
         authProvider
-          .filterUserVisibleWithProjectFallback(ds, userInfo)
+          .filterResourceProjectVisible(ds, userInfo)
       }
       _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done checking Sam permission")))
       res = samVisibleDisksOpt match {
         case None => Vector.empty
         case Some(samVisibleDisks) =>
           val samVisibleDisksSet = samVisibleDisks.toSet
-          // Making the assumption that users will always be able to access disks that they create
           disks
             .filter(d =>
-              d.auditInfo.creator == userInfo.userEmail || samVisibleDisksSet.contains(
+              samVisibleDisksSet.contains(
```

Comment:
in what situation would you have a disk in Leo and not in Sam? if it errored on creation or something? I think in that case the user wouldn't be able to see the disk because it wouldn't be in the permissions from the sam check. If the user was the project owner they could see it, but not at the reader level (but I would have to check)

Reply:
I'm not sure you can get into this state with Disks, but you definitely can with Runtimes. That's why the getRuntime method has the ugly check for user-is-creator; Leo keeps the record around after Sam has no more knowledge of it

Reply:
hmmm that's a case I didn't know about, so when we notify a resource as deleted to Sam it removes it? In that case the time I could see this being an issue is a project reader using the `includeDeleted` flag and not seeing previous runtimes they've created and deleted

Reply:
Yeah - although this might not apply to v1 resources that Leo manages directly through Google's API. I know that for WSM-mediated resources, WSM cleans up (incl. removing from Sam) and Leo retains the record, which will now scan in Sam as forbidden to the creator.